src/common/WorkQueue.h

```diff
@@ -598,6 +598,12 @@ class ShardedThreadPool {
     virtual void return_waiting_threads() = 0;
     virtual void stop_return_waiting_threads() = 0;
     virtual bool is_shard_empty(uint32_t thread_index) = 0;
+    void set_timeout(time_t ti) {
+      timeout_interval = ceph::make_timespan(ti);
```

Comment:
Perhaps these intervals (timeout_interval/suicide_interval) worth wrapping with std::atomic to make sure they're accessed properly from multiple threads...

Reply:
They could be accessed from different threads indeed. However changing the timeout configs should be infrequent, while timer was check by heartbeat frequently. So maybe it's ok to lost update once or so.

Reply:
Right but IMO it's better to wrap these vars with atomic and don't care about any side effects that might occur overwise anymore...

Reply:
WIP